ConsoleApp/Figure.cs

```diff
@@ -0,0 +1,58 @@
+namespace ConsoleApp
+{
+    public class Figure
+    {
+        protected string Type {get;set;}
+        protected double Area {get;set;}
+        protected int SidesNumber {get;set;}
+        protected double Perimeter {get;set;}
+
+        protected virtual void SetArea()
+        {
+            Area = 0;
+        }
+        protected virtual void SetPerimeter()
+        {
+            Perimeter = 0;
+        }
+
+        public Figure()
+        {
+            Type = "unknown";
+            SetArea();
+            SetPerimeter();
+            SidesNumber = 0;
+        }
+
+        public Figure(string type, int sidesNumber, double area, double perimeter)
```

Comment:
for this task, it's not critical, but for the future, don't calculate all properties in the constructor. it might be that you won't need them all